### ISSUE TITLE
dealing with multi-line prompts

### ISSUE THREAD

**Issue by wisotzky** [ktbyers/netmiko#551]
Kirk,
Could you give a recommendation, how to deal with devices having a multi-line command prompt?
Specific example:
```

[gl:configure system security]
A:admin@pe1#
```
In this case the command prompt is always 3 lines...
LINE#1: CRLF (empty line)
LINE#2: provides info about configuration mode and context in square brackets
LINE#3: Info about username and system name

To make functions like check_config_mode() happy, it would be required to parse LINE#2. But cross-checking methods like find_prompt() in base_connection.py - seems in case of multi-line prompts only the last line is considered. Would it be required and reasonable to overwrite the find_prompt() method?

**Comment by ktbyers** [comment]
@wisotzky Which platform are you working on?

In Juniper, I just look for the ']' to verify if I am in config mode:

https://github.com/ktbyers/netmiko/blob/develop/netmiko/juniper/juniper_ssh.py#L59

```
pyclass@pynet-jnpr-srx1> 

pyclass@pynet-jnpr-srx1> configure 
Entering configuration mode

[edit]
pyclass@pynet-jnpr-srx1# 
```

The find_prompt() method is inherently finding only the last line there i.e. `A:admin@pe1#`. I don't think I would want multiline prompt in Netmiko as it would be a departure from how it is done everywhere else (and I somehwat doubt it is necessary).

Note, pattern by default will use the trailing prompt to determine whether Netmiko is done reading.

```
    def check_config_mode(self, check_string='', pattern=''):
        """Checks if the device is in configuration mode or not."""
        log.debug("pattern: {0}".format(pattern))
        self.write_channel('\n')
        output = self.read_until_pattern(pattern=pattern)
        log.debug("check_config_mode: {0}".format(repr(output)))
        return check_string in output
```

So I think you could do something like the following:

```
    def check_config_mode(self, check_string='configure'):
        """Checks if the device is in configuration mode or not."""
        return super(YourClass, self).check_config_mode(check_string=check_string)
```

In other words you know you are in configuration mode if the 'configure' string is returned when you hit enter.

Let me know if that doesn't make sense or doesn't work for some reason.
